tests/CommonExtractor.php

```diff
@@ -0,0 +1,497 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Tests;
+
+use Keboola\Component\UserException;
+use Keboola\Csv\CsvWriter;
+use Keboola\Csv\Exception as CsvException;
+use Keboola\DbExtractorCommon\Configuration\BaseExtractorConfig;
+use Keboola\DbExtractorCommon\Configuration\DatabaseParameters;
+use Keboola\DbExtractorCommon\Configuration\TableDetailParameters;
+use Keboola\DbExtractorCommon\Configuration\TableParameters;
+use Keboola\DbExtractorCommon\Exception\ApplicationException;
+use Keboola\DbExtractorCommon\Exception\DeadConnectionException;
+use Keboola\DbExtractorCommon\BaseExtractor;
+use Keboola\DbExtractorCommon\RetryProxy;
+
+class CommonExtractor extends BaseExtractor
+{
+    public const TYPE_AUTO_INCREMENT = 'autoIncrement';
+    public const TYPE_TIMESTAMP = 'timestamp';
+
+    /** @var \PDO|null */
+    private $connection;
+
+    /** @var array */
+    private $incrementalFetching;
+
+    public function extract(BaseExtractorConfig $config): array
+    {
+        $imported = [];
+        $outputState = [];
+        if ($config->isConfigRow()) {
+            $tableParameters = TableParameters::fromRaw($config->getParameters());
```

Comment:
Tohle by klidně mohlo být rovnou v configu... vč. té kontroly toho, že je to fakt ConfigRow